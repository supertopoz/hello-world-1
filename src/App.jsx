import React from 'react';
//import './styles.css';


class App extends React.Component{
  constructor(props) {
  	super(props);
  this.state = {
    gifts : []
  }  
  this.addGift = this.addGift.bind(this)
  }

  addGift(){
    const { gifts } = this.state;
    const ids = this.state.gifts.map(gift => gift.id);
    const max_id = ids.length > 0 ? Math.max(...ids) : 0;
    gifts.push({id: max_id + 1});

    this.setState({ gifts })
    console.log(...this.state.gifts)
  }
 
  render(){
  	return(
  		<div>
  		<h1>Hello World 1</h1>
      <div className='btn-add' onClick={this.addGift}>Add Gift</div>
  		</div>
  	)
  }
}

export default App;




  