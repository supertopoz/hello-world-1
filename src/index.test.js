import React from 'react';
import App from './App';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow }  from 'enzyme';

Enzyme.configure({adapter: new Adapter()})

const app = shallow(<App />);

it('renders correctly', () => {
	expect(app).toMatchSnapshot();
})

it('int the `state` with an empty list', ()=> {
	expect(app.state().gifts).toEqual([]);
})

it('adds a new gift to `state` when clicking the `add gift` button', () => {
  app.find('.btn-add').simulate('click');
  expect(app.state().gifts).toEqual([{ id: 1 }])
})

it('renders a new gift to the dom',() => {
 //
})

