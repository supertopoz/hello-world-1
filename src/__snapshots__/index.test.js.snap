// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Hello World 1
        </h1>,
        <div
          className="btn-add"
          onClick={[Function]}
        >
          Add Gift
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello World 1",
        },
        "ref": null,
        "rendered": "Hello World 1",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Gift",
          "className": "btn-add",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add Gift",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Hello World 1
          </h1>,
          <div
            className="btn-add"
            onClick={[Function]}
          >
            Add Gift
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello World 1",
          },
          "ref": null,
          "rendered": "Hello World 1",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Gift",
            "className": "btn-add",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add Gift",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
